cmd/main.go

```diff
@@ -2,161 +2,52 @@ package main
-}
-
-func (w *FileWatcher) Update(path string, namespace string, podname string, containername string) error {
-	var add float64
-	var lastSize float64
-	var size float64
-
-	counter, err := w.metrics.GetMetricWithLabelValues(path, namespace, podname, containername)
-	if err != nil {
-		return err
-	}
-	stat, err := os.Stat(path)
-	if err != nil {
-		return err
-	}
-	if stat.IsDir() {
-		return nil // Ignore directories
-	}
-	lastSize, size = w.sizes[path], float64(stat.Size())
-	w.sizes[path] = size
-	if size > lastSize {
-		// File has grown, add the difference to the counter.
-		add = size - lastSize
-	} else if size < lastSize {
-		// File truncated, starting over. Add the size.
-		add = size
-	}
-	log.V(3).Info("For logfile in...", "path", path, "lastsize", lastSize, "currentsize", size, "addedbytes", add)
-	counter.Add(add)
-	return nil
-}
-
-func (w *FileWatcher) Watch() {
-
-	for {
-		//All logfiles with containername are added to the watcher
-		//write event for these logfiles are being watched
-		//create event gets issued for all new logfiles appear under logfilepathname /var/log/containers/
-		//For the cases new log files added, old files moved, old files deleted, you need to add/remove them from watcher as whole dir added to the watcher
-		//For new log files added write event is not getting issued
-
-		e, err := w.watcher.Event()
-		if err != nil {
-			log.Error(err, "Watcher.Event returning err")
-			os.Exit(1)
-		}
-
-		log.V(3).Info("Events notified for...", "e.Name", e.Name, "Event", e.Op)
-
-		//Get namespace, podname, containername from e.Name - log file path
-
-		r2 := kubernetesregexpCompiled.FindStringSubmatch(e.Name)
-
-		//if submatches == nil {
-		if r2 == nil {
-			log.V(2).Info("filename doesn't conform with k8 logfile path name ...", "filename", e.Name)
-		} else {
-			podname := r2[PodNameIndex]
-			namespace := r2[NamespaceIndex]
-			containername := r2[ContainerNameIndex]
-			dockerid := r2[DockerIndex]
-			log.V(3).Info("Namespace podname containername...", "namespace", namespace, "podname", podname, "containername", containername, "dockerid", dockerid)
-
-			err := w.Update(e.Name, namespace, podname, containername)
-			if err != nil {
-				log.V(2).Info("file e.Name Stat can't be checked", "filename", e.Name)
-			}
-		}
-
-	}
-}
-
 func main() {
-	var dir string
-	var addr string
-	var crtFile string
-	var keyFile string
-
-	//directory to be watched out where symlinks to all logs files are present e.g. /var/log/containers/
-	//debug option true or false
-	//listening port where this go-app push prometheus registered metrics for further collected or reading by end prometheus server
-	flag.StringVar(&dir, "dir", "/var/log/containers/", "Directory containing log files")
-	flag.IntVar(&verbosity, "verbosity", 0, "set verbosity level")
+	var (
+		dir       string
+		addr      string
+		crtFile   string
+		keyFile   string
+		verbosity int
+	)
+	flag.StringVar(&dir, "dir", logDir, "Directory containing log files")
```

Comment:
nit as not worth blocking this change to me but I don't see the point in making this configurable.  The actual subdirectory structure and the manor in which we wrote the regex is highly dependent upon a well known, likely to not change, pattern of how kube writes logs.  The root of which will likely always be /var/log/pods.  We may as well simply make the assumption and figure out how to evolve if needed in the future

Reply:
The reason it's configurable is purely to support the functional test that verifies we are publishing correctly formed metrics. It allows the exporter to monitor a test directory instead of /var/log/pods to verify the metrics published.